paddlenlp/experimental/transformers/fused_transformer_layers.py

```diff
@@ -1884,130 +2036,1059 @@ def __init__(self, config: FusedMultiTransformerConfig):
+                )
+                return query, key, value
+            else:
+                qkv_out = paddle.concat(
+                    [
+                        query.reshape([-1, self.num_heads * self.config.mla_config.qk_head_dim]),
+                        key.reshape([-1, self.num_heads * self.config.mla_config.qk_head_dim]),
+                        value.reshape([-1, self.num_heads * self.config.mla_config.v_head_dim]),
+                    ],
+                    axis=-1,
+                )
+                return qkv_out
+        else:
+            qkv_out = weight_only_linear(
+                ln_out,
+                weight=self.qkv_weights[i],
+                bias=self.qkv_biases[i],
+                weight_scale=self.qkv_weights_scale[i],
+                weight_dtype=self.weight_dtype,
+                group_size=self.weightonly_group_size,
+            )
+            return qkv_out
+
+    def compute_out_linear(self, fmha_out, i):
+        return weight_only_linear(
+            fmha_out,
+            weight=self.linear_weights[i],
+            weight_scale=self.linear_weights_scale[i],
+            weight_dtype=self.weight_dtype,
+            group_size=self.weightonly_group_size,
+        )
+
+    def compute_ffn1(self, tmp_out, i):
+        return weight_only_linear(
+            tmp_out,
+            weight=self.ffn1_weights[i],
+            weight_scale=self.ffn1_weights_scale[i],
+            weight_dtype=self.weight_dtype,
+            group_size=self.weightonly_group_size,
+        )
+
+    def compute_ffn2(self, ffn1_out, i):
+        return weight_only_linear(
+            ffn1_out,
+            weight=self.ffn2_weights[i],
+            weight_scale=self.ffn2_weights_scale[i],
+            weight_dtype=self.weight_dtype,
+            group_size=self.weightonly_group_size,
+        )
+
+    def compute_shared_expert(self, tmp_out, i):
+        ffn1_out = weight_only_linear(
+            tmp_out,
+            weight=self.shared_expert_ffn1_weights[i],
+            weight_scale=self.shared_expert_ffn1_weights_scale[i],
+            weight_dtype=self.weight_dtype,
+            group_size=self.weightonly_group_size,
+        )
+        ffn1_out = fused_bias_act(ffn1_out, None, act_method=self.activation)
+        ffn2_out = weight_only_linear(
+            ffn1_out,
+            weight=self.shared_expert_ffn2_weights[i],
+            weight_scale=self.shared_expert_ffn2_weights_scale[i],
+            weight_dtype=self.weight_dtype,
+            group_size=self.weightonly_group_size,
+        )
+        if self.config.moe_config.shared_expert_with_gate:
+            gate_out = paddle.matmul(tmp_out, self.shared_expert_gate_weights[i])
+            gate_out = paddle.nn.functional.sigmoid(gate_out)
+            return gate_out * ffn2_out
+        return ffn2_out
+
+
+class FusedMultiTransformerWINTX(FusedMultiTransformerBase):
```

Comment:
这个类和已有的类共性比较大，是不是可以基于已有的Transformer类来实现？
```suggestion
class FusedMultiTransformerWINTX(FusedMultiTransformerBase):
```

Reply:
已有的类最相似的`FusedMultiTransformerWeightOnly`。基于这个改的话需要改动的地方比较多。主要有以下问题：
1. FusedMultiTransformerWeightOnly不支持灵活的混合bit配置
2. weight shape设置与当前已导出权重差别比较大